expression/builtin_other_test.go

```diff
@@ -71,6 +74,55 @@ func (s *testEvaluatorSuite) TestBitCount(c *C) {
 	}
 }
 
+func (s *testEvaluatorSuite) TestInFunc(c *C) {
+	defer testleak.AfterTest(c)()
+	fc := funcs[ast.In]
+	time1 := types.Time{Time: types.FromGoTime(time.Now()), Fsp: 6, Type: mysql.TypeDatetime}
+	time2 := types.Time{Time: types.FromGoTime(time.Now()), Fsp: 6, Type: mysql.TypeDatetime}
+	time3 := types.Time{Time: types.FromGoTime(time.Now()), Fsp: 6, Type: mysql.TypeDatetime}
+	time4 := types.Time{Time: types.FromGoTime(time.Now()), Fsp: 6, Type: mysql.TypeDatetime}
+	duration1 := types.Duration{Duration: time.Duration(12*time.Hour + 1*time.Minute + 1*time.Second)}
+	duration2 := types.Duration{Duration: time.Duration(12*time.Hour + 1*time.Minute)}
+	duration3 := types.Duration{Duration: time.Duration(12*time.Hour + 1*time.Second)}
+	duration4 := types.Duration{Duration: time.Duration(12 * time.Hour)}
+	json1 := json.CreateJSON("123")
+	json2 := json.CreateJSON("123.1")
+	json3 := json.CreateJSON("123.2")
+	json4 := json.CreateJSON("123.3")
+	testCases := []struct {
+		args []interface{}
+		res  interface{}
+	}{
+		{[]interface{}{1, 1, 2, 3}, int64(1)},
+		{[]interface{}{1, 0, 2, 3}, int64(0)},
+		{[]interface{}{1, nil, 2, 3}, nil},
+		{[]interface{}{nil, nil, 2, 3}, nil},
+		{[]interface{}{uint64(0), 0, 2, 3}, int64(1)},
+		{[]interface{}{uint64(math.MaxUint64), uint64(math.MaxUint64), 2, 3}, int64(1)},
+		{[]interface{}{-1, uint64(math.MaxUint64), 2, 3}, int64(0)},
+		{[]interface{}{uint64(math.MaxUint64), -1, 2, 3}, int64(0)},
+		{[]interface{}{1, 0, 2, 3}, int64(0)},
+		{[]interface{}{1.1, 1.2, 1.3}, int64(0)},
+		{[]interface{}{1.1, 1.1, 1.2, 1.3}, int64(1)},
+		{[]interface{}{"1.1", "1.1", "1.2", "1.3"}, int64(1)},
+		{[]interface{}{"1.1", hack.Slice("1.1"), "1.2", "1.3"}, int64(1)},
+		{[]interface{}{hack.Slice("1.1"), "1.1", "1.2", "1.3"}, int64(1)},
+		{[]interface{}{time1, time2, time3, time1}, int64(1)},
+		{[]interface{}{time1, time2, time3, time4}, int64(0)},
+		{[]interface{}{duration1, duration2, duration3, duration4}, int64(0)},
+		{[]interface{}{duration1, duration2, duration1, duration4}, int64(1)},
+		{[]interface{}{json1, json2, json3, json4}, int64(0)},
+		{[]interface{}{json1, json1, json3, json4}, int64(1)},
+	}
+	for _, tc := range testCases {
+		fn, err := fc.getFunction(s.ctx, s.datumsToConstants(types.MakeDatums(tc.args...)))
+		c.Assert(err, IsNil)
+		d, err := evalBuiltinFunc(fn, types.DatumRow(types.MakeDatums(tc.args...)))
```

Comment:
We'd better not to call `evalBuiltinFunc` to write unit tests, you can take a look at `TestArithmeticPlus()` in file **expression/builtin_arithmetic_test.go** as an example.

Reply:
in `builtin_arithmtic_test.go`, TestArithmeticMod/Divide/Multiply and so on have all used `evalBuiltinFunc`.